### PR TITLE
[OV JS] Add human-pose-estimation sample

### PR DESCRIPTION
**Pull request by vishniakov-nikolai**
### Details:
 - Add human-pose-estimation sample for js bindings

### Tickets:
 - 141204

### CODE REVIEW

File: package-lock.json
Comment (ilya-lavrenov) [inline package-lock.json:1]: do we need it? 
`wasm-openvino` looks unsupported for now.

Comment (vishniakov-nikolai) [reply package-lock.json:1]: Was added by accident. Thank you!

File: samples/js/node/notebooks/pose-estimation.nnb [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+            "language": "markdown",
+            "source": [
+                "## Load the Model"
+            ],
+            "outputs": []
+        },
+        {
+            "language": "typescript",
+            "source": [
+                "const core = new ov.Core();\nconst model = await core.readModel(modelXMLPath);\nconst compiledModel = await core.compileModel(model, 'CPU', { PERFORMANCE_HINT: 'LATENCY' });\n\nconst inputLayer = compiledModel.inputs[0];\nconst outputLayers = compiledModel.outputs;\n\nconst [height, width] = inputLayer.shape.slice(2);\n\nconst pafsOutputKey = 'Mconv7_stage2_L1';\nconst heatmapsOutputKey = 'Mconv7_stage2_L2';\n\nconst THRESHOLD = 0.3;\nconst COLOR = [0, 255, 0, 255];"
```
Comment (almilosz) [inline samples/js/node/notebooks/pose-estimation.nnb:63]:
```suggestion
                "const core = new ov.Core();\nconst model = await core.readModel(modelXMLPath);\nconst compiledModel = await core.compileModel(model, 'CPU', { PERFORMANCE_HINT: 'LATENCY' });\n\nconst inputLayer = compiledModel.inputs[0];\nconst outputLayers = compiledModel.outputs;\n\nconst [height, width] = inputLayer.shape.slice(2);\n\nconst pathsOutputKey = 'Mconv7_stage2_L1';\nconst heatmapsOutputKey = 'Mconv7_stage2_L2';\n\nconst THRESHOLD = 0.3;\nconst COLOR = [0, 255, 0, 255];"
```
What does this variable contain?

File: samples/js/node/notebooks/pose-estimation.nnb [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+            "source": [
+                "## Process result"
+            ],
+            "outputs": []
+        },
+        {
+            "language": "typescript",
+            "source": [
+                "try {\n  const outputHeatmapShape = heatmaps.getShape();\n\n  // Define constants\n  const numKeypoints = 18;\n  const heatmapWidth = outputHeatmapShape[3];\n  const heatmapHeight = outputHeatmapShape[2];\n\n  // Step 1: Extract keypoints from heatmaps\n  let keypoints = [];\n  const heatmapsTFTensors = toTFTensor(heatmaps);\n  const xCoef = originalWidth / heatmapWidth;\n  const yCoef = originalHeight / heatmapHeight;\n\n  for (let i = 0; i < numKeypoints; i++) {\n    const heatmap = heatmapsTFTensors.slice(\n      [0, i, 0, 0],\n      [1, 1, heatmapHeight, heatmapWidth]).squeeze();\n\n    const { x, y, confidence } = getCoords(heatmap);\n\n    keypoints.push({\n      x: x * xCoef,\n      y: y * yCoef,\n      confidence,\n    });\n  }\n\n  drawKeypoints(originalImage, keypoints);\n  drawSkeleton(originalImage, keypoints);\n\n  displayArrayAsImage(originalImage.data, originalWidth, originalHeight, display);\n} catch(e) {\n  console.log(e);\n}"
```
Comment (almilosz) [inline samples/js/node/notebooks/pose-estimation.nnb:119]: Step one is below step 2 ;)

File: samples/js/node/notebooks/pose-estimation.nnb [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+            "language": "markdown",
+            "source": [
+                "## Do Inference"
+            ],
+            "outputs": []
+        },
+        {
+            "language": "typescript",
+            "source": [
+                "const tensorData = new Float32Array(inputImage);\nconst tensor = new ov.Tensor(ov.element.f32, inputLayer.shape, tensorData);\n\nconst inferRequest = compiledModel.createInferRequest();\ninferRequest.setInputTensor(tensor);\nconst outputs = inferRequest.infer();\n\nconst pafs = outputs[pafsOutputKey];\nconst heatmaps = outputs[heatmapsOutputKey];"
```
Comment (almilosz) [inline samples/js/node/notebooks/pose-estimation.nnb:105]: If I see correctly this model operates on FP16-INT8. Didn't you have any problems with it?

Comment (almilosz) [inline samples/js/node/notebooks/pose-estimation.nnb:105]: `const pafs = outputs[pafsOutputKey]`
What this output contain? It is not used later.

**Review by almilosz — COMMENTED** [review]
- the notebook name does not contain the number like other notebooks have. Those numbers were removed from python notebooks. Maybe we should also do it?
- list of notebooks should be updated -> _samples/js/node/README.md_
- node_modules/braces 7.0.1 is marked as high severity vulnerability.  `npm audit fix`?

File: samples/js/node/notebooks/pose-estimation.nnb (hunk 1 above)
Comment (vishniakov-nikolai) [reply samples/js/node/notebooks/pose-estimation.nnb:63]: Which one? :)

File: samples/js/node/notebooks/pose-estimation.nnb [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
...Index % tensor.shape[1];\n  const y = Math.floor(maxIndex / tensor.shape[1]);\n\n  return { x, y, confidence: maxVal };\n}\n\n// Draw keypoints on the input image\nfunction drawKeypoints(image, keypoints) {\n  keypoints.forEach(keypoint => {\n    if (keypoint.confidence > THRESHOLD) {\n      cv.circle(image,\n        new cv.Point(keypoint.x, keypoint.y),\n        2,\n        COLOR,\n        3,\n      );\n    }\n  });\n}\n\n// Draw skeleton (lines between keypoints) on the input image\nfunction drawSkeleton(image, keypoints) {\n  const skeleton = [\n    [1, 2], [1, 5], [2, 3], [3, 4], [5, 6], [6, 7], [1, 8], [8, 9], [9, 10],\n    [1, 11], [11, 12], [12, 13], [1, 0], [0, 14], [14, 16], [0, 15], [15, 17],\n  ];\n\n  skeleton.forEach(([start, end]) => {\n    const startPoint = keypoints[start];\n    const endPoint = keypoints[end];\n\n    if (startPoint.confidence > THRESHOLD\n      && endPoint.confidence > THRESHOLD) {\n      cv.line(image,\n        new cv.Point(startPoint.x, startPoint.y),\n        new cv.Point(endPoint.x, endPoint.y),\n        COLOR,\n        2,\n      );\n    }\n  });\n}\n\nfunction toTFTensor(ovTensor) {\n  return tf.tensor(ovTensor.data, ovTensor.getShape());\n}"
```
Comment (almilosz) [inline samples/js/node/notebooks/pose-estimation.nnb:91]: Can't we extract this info from the model output? 
`const skeleton = [
    [1, 2], [1, 5], [2, 3], [3, 4], [5, 6], [6, 7], [1, 8], [8, 9], [9, 10],
    [1, 11], [11, 12], [12, 13], [1, 0], [0, 14], [14, 16], [0, 15], [15, 17],
  ];`

File: samples/js/node/notebooks/pose-estimation.nnb (hunk 1 above)
Comment (almilosz) [reply samples/js/node/notebooks/pose-estimation.nnb]: pafsOutputKey
From the [docs](https://docs.openvino.ai/2022.3/omz_models_model_human_pose_estimation_0001.html): 
> Name: Mconv7_stage2_L1, shape: 1, 38, 32, 57 contains keypoint pairwise relations (part affinity fields).

File: samples/js/node/notebooks/pose-estimation.nnb (hunk 4 above)
Comment (almilosz) [reply samples/js/node/notebooks/pose-estimation.nnb:91]: discussed offline

**Review by almilosz — APPROVED** [review]

**Pull request merged**